config/config.go

```diff
@@ -364,13 +373,27 @@ func ParseUpstream(upstream string) (Upstream, error) {
 	}
 
 	return Upstream{
-		Net:  n,
-		Host: host,
-		Port: port,
-		Path: path,
+		Net:        n,
+		Host:       host,
+		Port:       port,
+		Path:       path,
+		CommonName: commonName,
 	}, nil
 }
 
+func extractCN(in string) (cn string, upstream string) {
+	hashIdx := strings.Index(in, "#")
+
+	if hashIdx >= 0 {
+		cn = strings.TrimPrefix(in[hashIdx:], "#")
+		upstream = in[:hashIdx]
+	} else {
+		upstream = in
+	}
+
+	return
```

Comment:
`strings.Cut` can simplify this:
```suggestion
	upstream, commonName, _ := strings.Cut(in, "#")
	return upstream, commonName
```

Reply:
Good call. I did reverse the order of return values from your suggestion for consistency with the other `extract*()` routines.